### PR TITLE
Increase some of the DnsEndPointTest timeouts

### LINKED ISSUE #57929: Tests time out: System.Net.Sockets.Tests.DnsEndPointTest - mostly on SLES.15 in release/6.0 branches
Failures:
- `Socket_ConnectAsyncDnsEndPoint_HostNotFound`
  - 6/24-8/23 (incl. PRs): **241** -- each day at least one except 8/4 and 8/7.
    - Failures are on release/6.0* branches (preview3+) -- only 1 failure on 7/16 is in main branch, no failure on any PR
    - OS failures:
      - SLES.15.Amd64.Open - **236x**
      - Windows.Nano.1809.Amd64.Open - 2x (1 occurrence on main branch - 7/16, 7/18)
      - Windows.10.Amd64.Server19H1.Open - 1x (7/30)
      - Windows.7.Amd64.Open - 1x (8/12)
      - Windows.10.Amd64.ServerRS5.Open - 1x (8/16)
  - 8/24-8/29 (incl. PRs): **15** - all release/6.0*, official builds, SLES.15.Amd64.Open
  - 8/30-9/5 (incl. PRs): **20** - all release/6.0*, 2x PR #58336, SLES.15.Amd64.Open
- `Socket_StaticConnectAsync_HostNotFound`
  - 6/24-8/23 (incl. PRs): **95** -- each day at least one except 7/7, 7/18, 8/5, 8/7-8/10, 8/12, 8/14-8/15, 8/17-8/18, 8/20.
    - All failures are on release/6.0* branches (preview3+)
    - OS failures:
      - SLES.15.Amd64.Open - **93x**
      - Windows.Nano.1809.Amd64.Open - 1x (7/17)
      - Windows.10.Amd64.Server19H1.Open - 1x (7/22)
  - 8/24-8/29 (incl. PRs): **9** -- all release/6.0*, official builds, SLES.15.Amd64.Open
  - 8/30-9/5 (incl. PRs): **12** -- 1x main=7.0 on Windows.10.Amd64.Server19H1.Open, otherwise release/6.0* on SLES.15.Amd64.Open, 2x PR #58336
- `Socket_StaticConnectAsync_Success` -- **main branch failures tracked in #21924**
  - 6/24-8/23 (incl. PRs): **21**
    - Mix of branches and OSs
      - 4x main + 1x PR + 16x release/6.0-preview* branches
      - OSs:
        - SLES.15.Amd64.Open - 4x
        - Debian.9.Arm32.Open - 4x
        - OSX.1015.Amd64.Open - 2x (incl. 1x Mono)
        - Ubuntu.1804.ArmArch.Open - 2x
        - Ubuntu.1910.Amd64.Open - 2x
        - Fedora.32.Amd64.Open - 3x
        - OSX.1013.Amd64.Open - 1x
        - Windows.10.Amd64.ServerRS5.Open - 1x
        - Debian.9.Amd64.Open - 1x
        - Windows.10.Amd64.Server19H1.Open - 1x
  - 8/24-8/29 (incl. PRs): **2** -- all release/6.0*, official builds -- 1x SLES.15.Amd64.Open, 1x Debian.9.Amd64.Open
  - 8/30-9/5 (incl. PRs): **1** -- main=7.0 official run on Ubuntu.1804.ArmArch.Open

Kusto query:
```kusto
let failedTests = (methodName : string, includePR : bool, messageSubstr: string) {
cluster('engsrvprod.kusto.windows.net').database('engineeringdata').TestResults
| where Type startswith "System.Net" 
    and Result == 'Fail'
| where Method == methodName
  or Method == 'Socket_StaticConnectAsync_Success'
  or Method == 'Socket_StaticConnectAsync_HostNotFound'
  or Method == 'Socket_ConnectAsyncDnsEndPoint_HostNotFound'
| where Message contains messageSubstr
| distinct JobId, WorkItemId, Message, Exception, StackTrace, Method, Type //, Arguments
| join kind=inner (cluster('engsrvprod.kusto.windows.net').database('engineeringdata').Jobs
    | where ((Branch == 'refs/heads/main') or (Branch == 'refs/heads/master') or (includePR and (Source startswith "pr/")) or (Branch startswith 'refs/heads/release/6.0'))
    | where Type startswith "test/functional/cli/"
        and not(Properties contains "runtime-staging")
    | summarize arg_max(Finished, Properties, Type, Branch, Source, Started, QueueName) by JobId
| project-rename JobType = Type) on JobId
};
failedTests('', true, '');
```

Failure:
```
Timed out while waiting for connection
Expected: True
Actual:   False

   at System.Net.Sockets.Tests.DnsEndPointTest.Socket_ConnectAsyncDnsEndPoint_HostNotFound() in /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/DnsEndPointTest.cs:line 275
```

### PR DESCRIPTION
**Pull request by antonfirsov**
Long story short:
I have spent several hours trying to get a test run on SLES  in order to repro #57929, no success so far, and it may take days to make progress, since I'm unfamiliar with SLES.

I recommend to bump the timeout values for now, and see if it helps with the issue. If not, we may invest into another round of investigation.

(hopefully) fixes #57929

### CODE REVIEW

**Comment by antonfirsov** [comment]
Reacting to https://github.com/dotnet/runtime/pull/58129#discussion_r696037908, I pushed a change that renames existing `TestSettings.PassingTestTimeout` to `PassingTestShortTimeout` and introducing `PassingTestLongTimeout = 30sec`. This is a necessary change to improve the maintainability for the code on `master`, backporting the entire change to 6.0 will make it easier to adapt future test improvements.

File: src/libraries/Common/tests/System/Net/Sockets/TestSettings.cs [hunk 1]
```diff
@@ -8,9 +8,10 @@ namespace System.Net.Sockets.Tests
     public static class TestSettings
     {
         // Timeout values in milliseconds.
-        public const int PassingTestTimeout = 10000;
```
Comment (wfurt) [inline src/libraries/Common/tests/System/Net/Sockets/TestSettings.cs:11]: Can you keep this untouched to be consistent with other tests? e.g. `PassingTestTimeout` is normal/default timeout. If we need anything different for come case we would make it with specific name - like `PassingTestLongTimeout` (that part looks ok to me)

**Comment by karelz** [comment]
[Test run](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-58129-merge-efc5621110af457ca4/System.Net.Http.Functional.Tests/1/console.e5ce7e56.log?sv=2019-07-07&se=2021-09-20T14%3A52%3A04Z&sr=c&sp=rl&sig=UR1f2Au3SGjr%2FEcH%2FM5%2B3MOn89kAtdP%2BLr7q0n17gT8%3D) reported networking test failures 4x this specific test:
```
System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_ServerCertificates_Test.UseCallback_ValidCertificate_ExpectedValuesDuringCallback(remoteServer: (BaseUri: https://corefx-net-http11.azurewebsites.net/, HttpVersion: 1.1), url: https://corefx-net-http11.azurewebsites.net/Echo.ashx, checkRevocation: True) [FAIL]

      System.Net.Http.HttpRequestException : The SSL connection could not be established, see inner exception.
      ---- Expected None, got RemoteCertificateChainErrors with chain status RevocationStatusUnknown, OfflineRevocation
      Expected: True
      Actual:   False
      Stack Trace:
        /_/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs(92,0): at System.Net.Http.ConnectHelper.EstablishSslConnectionAsync(SslClientAuthenticationOptions sslOptions, HttpRequestMessage request, Boolean async, Stream stream, CancellationToken cancellationToken)
        /_/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs(1364,0): at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
        /_/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs(1430,0): at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
        /_/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs(454,0): at System.Net.Http.HttpConnectionPool.AddHttp11ConnectionAsync(HttpRequestMessage request)
        /_/src/libraries/Common/src/System/Threading/Tasks/TaskCompletionSourceWithCancellation.cs(21,0): at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)
        /_/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs(548,0): at System.Net.Http.HttpConnectionPool.GetHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
        /_/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs(988,0): at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
        /_/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/RedirectHandler.cs(30,0): at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
        /_/src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTestBase.cs(119,0): at System.Net.Http.Functional.Tests.HttpClientHandlerTestBase.VersionCheckerHttpHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
        /_/src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs(527,0): at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
        /_/src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.ServerCertificates.cs(173,0): at System.Net.Http.Functional.Tests.HttpClientHandler_ServerCertificates_Test.UseCallback_ValidCertificate_ExpectedValuesDuringCallback(RemoteServer remoteServer, Uri url, Boolean checkRevocation)
        --- End of stack trace from previous location ---
        ----- Inner Stack Trace -----
        /_/src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.ServerCertificates.cs(159,0): at System.Net.Http.Functional.Tests.HttpClientHandler_ServerCertificates_Test.<>c__DisplayClass14_1.<UseCallback_ValidCertificate_ExpectedValuesDuringCallback>b__0(HttpRequestMessage request, X509Certificate2 cert, X509Chain chain, SslPolicyErrors errors)
        /_/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs(49,0): at System.Net.Http.ConnectHelper.<>c__DisplayClass1_0.<SetUpRemoteCertificateValidationCallback>b__0(Object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors)
        /_/src/libraries/System.Net.Security/src/System/Net/Security/SecureChannel.cs(998,0): at System.Net.Security.SecureChannel.VerifyRemoteCertificate(RemoteCertificateValidationCallback remoteCertValidationCallback, SslCertificateTrust trust, ProtocolToken& alertToken, SslPolicyErrors& sslPolicyErrors, X509ChainStatusFlags& chainStatus)
        /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs(639,0): at System.Net.Security.SslStream.CompleteHandshake(ProtocolToken& alertToken, SslPolicyErrors& sslPolicyErrors, X509ChainStatusFlags& chainStatus)
        /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs(652,0): at System.Net.Security.SslStream.CompleteHandshake(SslAuthenticationOptions sslAuthenticationOptions)
        /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs(435,0): at System.Net.Security.SslStream.ForceAuthenticationAsync[TIOAdapter](TIOAdapter adapter, Boolean receiveFirst, Byte[] reAuthenticationData, Boolean isApm)
        /_/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs(68,0): at System.Net.Http.ConnectHelper.EstablishSslConnectionAsync(SslClientAuthenticationOptions sslOptions, HttpRequestMessage request, Boolean async, Stream stream, CancellationToken cancellationToken)
```

That may be just some networking hiccup I guess ... cc @wfurt 

Rerunning the failed legs.

File: src/libraries/Common/tests/System/Net/Sockets/TestSettings.cs (hunk 1 above)
Comment (antonfirsov) [reply src/libraries/Common/tests/System/Net/Sockets/TestSettings.cs:11]: Reverted this renaming to reduce the diff size.

I think more verbosity in the names wouldn't hurt here. 10 sec is quite a short time in comparison to Http.Tests's default `PassingTestTimeout=30sec`. We use hardcoded 30 seconds at a bunch of places in socket tests, which we will eventually replace with `PassingTestLongTimeout`, meaning that the "default" `PassingTestTimeout` will become much less common.

**Review by wfurt — APPROVED** [review]
LGTM

**Comment by karelz** [comment]
Test failures:
- `BeginReceiveFromV4BoundToSpecificV4_Success` timeout - likely unrelated - @antonfirsov can you please confirm?
```
    System.Net.Sockets.Tests.DualModeConnectionlessBeginReceiveFrom.BeginReceiveFromV4BoundToSpecificV4_Success [FAIL]
      System.TimeoutException : The operation has timed out.
      Stack Trace:
        /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/DualModeSocketTest.cs(1612,0): at System.Net.Sockets.Tests.DualModeConnectionlessBeginReceiveFrom.BeginReceiveFrom_Helper(IPAddress listenOn, IPAddress connectTo, Boolean expectedToTimeout)
        /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/DualModeSocketTest.cs(1525,0): at System.Net.Sockets.Tests.DualModeConnectionlessBeginReceiveFrom.BeginReceiveFromV4BoundToSpecificV4_Success()
```
- Azure storage outage? -- `.packages/microsoft.dotnet.helix.sdk/6.0.0-beta.21427.6/tools/Microsoft.DotNet.Helix.Sdk.MonoQueue.targets(55,5): error : (NETCORE_ENGINEERING_TELEMETRY=Helix) StorageException: nodename nor servname provided, or not known (helixde8s23ayyeko0k025g8.blob.core.windows.net:443)` ([logs](https://github.com/dotnet/runtime/pull/58129/checks?check_run_id=3485526338))
- 2x System.Runtime.Tests killed - unrelated failures - [Debian.11](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-58129-merge-fbebc3b38a014ca890/System.Runtime.Tests/1/console.d6aceea7.log?sv=2019-07-07&se=2021-09-21T14%3A35%3A33Z&sr=c&sp=rl&sig=iWNfuR7JTB5idUbMgy1q6l3qg6FD%2BvRO6Nv%2BeH%2FUTdo%3D) and [Debian.10](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-58129-merge-7879db9a3b92427ab8/System.Runtime.Tests/1/console.f0358a02.log?sv=2019-07-07&se=2021-09-21T14%3A35%3A33Z&sr=c&sp=rl&sig=XvGm0sagbJj7SnMVNgprNXzBGzOxIC9T8HH4u4qAmsY%3D)
```
/root/helix/work/workitem /root/helix/work/workitem
  Discovering: System.Runtime.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Runtime.Tests (found 28 of 6264 test cases)
  Starting:    System.Runtime.Tests (parallel test collections = on, max threads = 2)
./RunTests.sh: line 162:    24 Killed
```

Rerunning tests.

**Comment by antonfirsov** [comment]
Yes, `BeginReceiveFromV4BoundToSpecificV4_Success ` is unrelated.

**Comment by karelz** [comment]
Same System.Runtime.Tests unrelated failures on [Debian.10](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-58129-merge-39558fa43af44f0e8b/System.Runtime.Tests/1/console.ae2d6ed3.log?sv=2019-07-07&se=2021-09-22T15%3A47%3A20Z&sr=c&sp=rl&sig=5Nk5bMhlVbAk0IufaLYVT%2BQksgUXJFT84TkQwqO4RL4%3D) and [Debian.11](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-58129-merge-40b8439ed03d48369c/System.Runtime.Tests/1/console.2fef811d.log?sv=2019-07-07&se=2021-09-22T15%3A47%3A21Z&sr=c&sp=rl&sig=JpvfVuQR8FxESD6MueiBe51QTKmzgdhp0uL6k8Ouos4%3D) - tracked by #58616:
```
  Discovering: System.Runtime.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Runtime.Tests (found 28 of 6264 test cases)
  Starting:    System.Runtime.Tests (parallel test collections = on, max threads = 2)
./RunTests.sh: line 162:    24 Killed
```

**Pull request merged**